goosebit/updates/swdesc/rauc.py

```diff
@@ -0,0 +1,42 @@
+import configparser
+import logging
+
+import semver
+from anyio import Path, open_file
+from PySquashfsImage import SquashFsImage
+
+from .func import sha1_hash_file
+
+MAGIC = b"hsqs"
+
+logger = logging.getLogger(__name__)
+
+
+async def parse_file(file: Path):
+    async with await open_file(file, "r+b") as f:
+        image_data = await f.read()
+
+        image = SquashFsImage.from_bytes(image_data)
+        manifest = image.select("manifest.raucm")
+        manifest_str = manifest.read_bytes().decode("utf-8")
+        config = configparser.ConfigParser()
+        config.read_string(manifest_str)
+        swdesc_attrs = parse_descriptor(config)
+
+        stat = await file.stat()
+        swdesc_attrs["size"] = stat.st_size
+        swdesc_attrs["hash"] = await sha1_hash_file(f)
+    return swdesc_attrs
+
+
+def parse_descriptor(manifest: configparser.ConfigParser):
+    swdesc_attrs = {}
+    try:
+        # specified as optional in the RAUC docs
+        swdesc_attrs["version"] = semver.Version.parse(manifest["update"].get("version") + ".0")
+        swdesc_attrs["compatibility"] = [{"hw_model": "default", "hw_revision": manifest["update"]["compatible"]}]
```

Comment:
Having a hard-coded hw_model "default" means that only a single hardware (platform) is supported to be managed in goosebit. Seems very limiting.